unittest/liegroups.cpp

```diff
@@ -490,6 +490,49 @@ struct LieGroup_JintegrateJdifference
   }
 };
 
+struct LieGroup_dIntegrateTransport
+{
+  template<typename T>
+  void operator()(const T) const
+  {
+    typedef typename T::ConfigVector_t ConfigVector_t;
+    typedef typename T::TangentVector_t TangentVector_t;
+    typedef typename T::JacobianMatrix_t JacobianMatrix_t;
+
+    PINOCCHIO_COMPILER_DIAGNOSTIC_PUSH
+    PINOCCHIO_COMPILER_DIAGNOSTIC_IGNORED_MAYBE_UNINITIALIZED
+    T lg;
+    BOOST_TEST_MESSAGE(lg.name());
+    ConfigVector_t qa, qb(lg.nq());
+    qa = lg.random();
+    TangentVector_t v(lg.nv()), tvec_at_qb(lg.nv()), tvec_at_qa(lg.nv()), tvec_at_qa_r(lg.nv());
+    v.setRandom();
+    lg.integrate(qa, v, qb);
+
+    // transport random tangent vector from q1 to q0
+    tvec_at_qb.setRandom();
+    lg.dIntegrateTransport(qa, v, tvec_at_qb, tvec_at_qa, ARG0);
+
+    // test reverse direction
+    TangentVector_t v_r = -v; // reverse path
+    ConfigVector_t qa_r = lg.integrate(qb, v_r);
+    lg.dIntegrateTransport(qa_r, v_r, tvec_at_qa, tvec_at_qa_r, ARG0);
```

Comment:
Shouldn't this be ``dIntegrateTransport(qb, v_r, tvec_at_qa, ...)``?

Since we check that ``qa`` is equal to ``qa_r`` below, this line amounts to integrating $v_r$ from $qa_r$, where we will end... Somewhere else than $q_a$ or $q_b$ :sweat_smile:

Meanwhile this line would fully make sense to me if we start from $q_b$: then $q_b \oplus v_r$ would land on $q_{ar} = q_a$, where we take ``tvec_at_qa``, and transport it back to $q_b$, which is why we have the equality between ``tvec_at_qb`` and ``tvec_at_qa_r`` below. (But then, ``tvec_at_qa_r`` lies in the tangent space at $q_b$, not $q_{ar}$! :stuck_out_tongue:)

@fabinsch I understand the test passed so I must have missed something. If you can double check at some point that would be :ok_hand:

Reply:
Meanwhile I fully agree with the Python variant.